core/arch/arm/mm/tee_mmu.c

```diff
@@ -293,7 +293,7 @@ TEE_Result vm_map_pad(struct user_ta_ctx *utc, vaddr_t *va, size_t len,
 	if (res)
 		goto err_rem_reg;
 
-	if (!(reg->attr & TEE_MATTR_PERMANENT) && mobj_is_paged(mobj)) {
+	if (mobj_is_paged(mobj)) {
```

Comment:
Do i understand this correctly: the check is not needed because a paged `mobj` is never supposed to have `TEE_MATTR_PERMANENT` set, right?

Reply:
That's correct. I can't recall why the test was added to begin with. Perhaps some extra precaution which was never removed, until now.